test/convenience/webhook.test.ts

```diff
@@ -1,4 +1,5 @@
 /// <reference types="npm:@types/node" />
+/// <reference types="npm:bun-types" />
```

Comment:
You may want to import this regularly instead of via reference, I'm not entirely sure if that's possible, but Node and Bun seem to have conflicting global types. 

Perhaps it's even better to import both things via import statement. Something like `const x: import("..").Bun = ..;` is what I have in mind. Hope it helps.

Reply:
> Perhaps it's even better to import both things via import statement. Something like const x: import("..").Bun = ..; is what I have in mind.

Hmm, my intellisense gives me the correct type but the tests still thinks the types are "any".

Reply:
I am thinking of using a type declaration, but it's not being picked up in Deno. Does Deno not support this?

Reply:
A type declaration? What do you mean? `type X = 0;`?

Reply:
Why do we need the types from bun and node in the first place? Can't we just model them in the same way we model everything else? Then we can get rid of triple-slash references entirely and everything will work everywhere.

Reply:
> Hmm, my intellisense gives me the correct type but the tests still thinks the types are "any".

this probably happens because your ide uses the ts lang server to resolve types, but deno doesn't. I recently faced exactly the same issue with different product and bun because of this
so another thing that can be done is to contribute to deno

Reply:
> Why do we need the types from bun and node in the first place? Can't we just model them in the same way we model everything else? Then we can get rid of triple-slash references entirely and everything will work everywhere.

I still think this is the way to go, it makes Node/Bun types completely superfluous

Reply:
> You don't need to contribute anything to Deno. You just have to use their LSP instead of tsserver.

um, how?
i meant that deno does not use the ts lang server to resolve types, so it has its own logic, which is less flexible, as we can see - ide can resolve bun's node-like imports, but deno cannot (most likely because of some special case only for imports starting from "node:")

Reply:
> this probably happens because your ide uses the ts lang server to resolve types, but deno doesn't. I recently faced exactly the same issue with different product and bun because of this

nvm, i forgot what the original problem was when wrote that

Reply:
> Why do we need the types from bun and node in the first place? Can't we just model them in the same way we model everything else? 

Because these are integration tests. The point is that we use the types from their exact implementation. It removes the purpose of the test if we were to stub them just like in our implementation. It's the same as testing code against itself.

Reply:
That is true, but as three pairs of eyes have reviewed this 3-line type now, I am fairly confident that we got it right :D

Also, Jarred cannot change it before Bun 2.0 as that would be breaking.  If we ever find a way to import the types in an isolated way without polluting the global types, we can improve this code in the future. For now, I don't want that to block us further.